lib/collection/src/collection_manager/segments_searcher.rs

```diff
@@ -174,21 +174,23 @@ impl SegmentsSearcher {
             let use_sampling =
                 sampling_enabled && segments.len() > 1 && available_points_segments > 0;
 
-            segments
-                .iter()
-                .map(|(_id, segment)| {
-                    (
-                        segment.clone(),
-                        search_in_segment(
-                            segment.clone(),
-                            batch_request.clone(),
-                            available_points_segments,
-                            use_sampling,
-                        ),
+            let mut searches: Vec<_> = vec![];
+            for (_id, segment) in segments.iter() {
+                let segment = segment.clone();
+                let segment_ret = segment.clone();
+                let batch_request = batch_request.clone();
+                let search = runtime_handle.spawn_blocking(move || {
+                    search_in_segment(
+                        segment,
+                        batch_request,
+                        available_points_segments,
+                        use_sampling,
                     )
-                })
-                .map(|(segment, f)| (segment, runtime_handle.spawn(f)))
-                .unzip()
+                });
+                searches.push((segment_ret, search));
+            }
+
+            searches.into_iter().unzip()
```

Comment:
What about keeping the iterator-style? Or do you like your approach better?

```rust
            segments
                .iter()
                .map(|(_id, segment)| {
                    let search = runtime_handle.spawn_blocking({
                        let (segment, batch_request) = (segment.clone(), batch_request.clone());
                        move || {
                            search_in_segment(
                                segment,
                                batch_request,
                                available_points_segments,
                                use_sampling,
                            )
                        }
                    });
                    (segment.clone(), search)
                })
                .unzip()
```

Reply:
Thanks for mentioning this, I used your suggestion. I like it better :+1: 

I had lifetime issues that I understood better by moving to a for-loop.